batch/batch.go

```diff
@@ -0,0 +1,96 @@
+package batch
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/coreos/bbolt"
+)
+
+// errSolo is a sentinel error indicating that the requester should re-run the
+// operation in isolation.
+var errSolo = errors.New(
+	"batch function returned an error and should be re-run solo",
+)
+
+type request struct {
+	*Request
+	errChan chan error
+}
+
+type batch struct {
+	db     *bbolt.DB
+	start  sync.Once
+	reqs   []*request
+	clear  func(b *batch)
```

Comment:
Would it be better to make this `clear func()`? The scheduler has the instance already and could use it in a lambda.